debugging/netcoreapp/prereqManager.ts

```diff
@@ -0,0 +1,93 @@
+import * as path from 'path';
+import * as vscode from 'vscode';
+import { BrowserClient } from './browserClient';
+import { MacNuGetPackageFallbackFolderPath } from './dockerManager';
+import { FileSystemProvider } from './fsProvider';
+import { OSProvider } from './osProvider';
+
+export interface Prerequisite {
+    checkPrerequisite(): Promise<boolean>;
+}
+
+type showErrorMessageFunction = (message: string, ...items: vscode.MessageItem[]) => Thenable<vscode.MessageItem>;
+
+export class DotNetExtensionInstalledPrerequisite implements Prerequisite {
+    constructor(
+        private readonly browserClient: BrowserClient,
+        // tslint:disable-next-line:no-any
+        private readonly getExtension: (extensionId: string) => vscode.Extension<any>,
+        private readonly showErrorMessage: showErrorMessageFunction) {
+    }
+
+    public async checkPrerequisite(): Promise<boolean> {
+        // NOTE: Debugging .NET Core in Docker containers requires the C# (i.e. .NET Core debugging) extension.
+        //       As Docker debugging is experimental, we don't want the extension as a whole to depend on it.
```

Comment:
We wouldn't want this even if it weren't experimental.  This is a generic docker extension, not just Azure.

Reply:
Fair point; updated comment.